### PR TITLE
refactor: move tkn-results into the main tree

### LINKED ISSUE #551: move `tkn-results` to the same tree as the results package
## Current State

`tkn-results` is currently in its module separate from the main tree

## Expected

moving `tkn-results` to the same tree will make it easier to publish releases as well as reduce the redundant packages while vendoring

### PR DESCRIPTION
**Pull request by avinal**
# Changes

move tkn-results to the main tree
    
- restruture the directories for tkn-results
- add vendoring for tkn-results
- delete tools/tkn-results
- fixes #551

improve tkn-results help docs
    
- add example for get and list
- regenerate cli docs
- add makefile entry for generating docs
- fix golangci-lint and goimports error
    
Signed-off-by: Avinal Kumar <avinal@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [x] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
action required: tkn-results has been moved into the main tree of tekton results
```

### CODE REVIEW

File: pkg/cli/client/client.go [hunk 1]
```diff
@@ -9,25 +9,29 @@ import (
...
 	"k8s.io/client-go/kubernetes"
+
+	// Load auth plugins
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-type ClientFactory struct {
+// Factory contains the configuration for creating a k8s client.
+type Factory struct {
```
Comment (khrm) [inline pkg/cli/client/client.go:28]: Nice naming. Following the convention. :)

**Review by khrm — COMMENTED** [review]
/approve

File: pkg/cli/client/client.go (hunk 1 above)
Comment (avinal) [reply pkg/cli/client/client.go:28]: Yeah, had to change a couple of names due to linting errors 😸

File: docs/cli/tkn-results_logs_list.md [hunk 2]
```diff
@@ -1,11 +1,24 @@
 ## tkn-results logs list
 
-List Logs
+List Logs for a given Result
 
+### Synopsis
+
+List Logs for a given Result. <result-name> is typically of format <namespace>/results/<parent-run-uuid>. '-' may be used as <result-name> to query all Logs for a given parent.
```
[the remark this answers is not in the document]
Comment (avinal) [reply docs/cli/tkn-results_logs_list.md:7]: Good catch, let me clarify it.

**Pull request merged**